test/odbc/test_pg_datatype_bit.cpp

```diff
@@ -0,0 +1,764 @@
+
+    for (int j = 0; j < operations_query.size(); j++) {
+      ASSERT_EQ(col_len[j], DATA_BYTES_EXPECTED);
+      ASSERT_EQ(col_results[j], expected_results[i][j]);
+    }
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_Bit, View_Creation) {
+  const string VIEW_QUERY = "SELECT * FROM " + TABLE_NAME;
+  const int PK_BYTES_EXPECTED = 4;
+  const int DATA_BYTES_EXPECTED = 1;
+
+  int pk;
+  unsigned char data;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector <string> valid_inserted_values = {
+    "0",
+    "1",
+    "NULL"
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns = {
+    {1, SQL_C_LONG, &pk, 0, &pk_len},
+    {2, SQL_C_BIT, &data, 0,  &data_len}
+  };
+
+  string insert_string{}; 
+  string comma{};
+  
+  for (int i = 0; i < valid_inserted_values.size(); i++) {
+    insert_string += comma + "(" + std::to_string(i) + "," + valid_inserted_values[i] + ")";
+    comma = ",";
+  }
+
+  // Create Table
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+ 
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(affected_rows, valid_inserted_values.size());
+  
+  odbcHandler.CloseStmt();
+
+  // Create view
+  odbcHandler.ExecQuery(CreateViewStatement(VIEW_NAME, VIEW_QUERY));
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(VIEW_NAME, {"*"}, vector<string> {COL1_NAME}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < valid_inserted_values.size(); i++) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+    ASSERT_EQ(pk_len, PK_BYTES_EXPECTED);
+    ASSERT_EQ(pk, i);
+
+    if (valid_inserted_values[i] != "NULL")
+    {
+      ASSERT_EQ(data_len, DATA_BYTES_EXPECTED);
+      ASSERT_EQ(data, StringToBit(valid_inserted_values[i]));
+    }
+    else 
+      ASSERT_EQ(data_len, SQL_NULL_DATA);
+  }
+
+  // Assert that there is no more data
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_Bit, Table_Unique_Constraints) {
```

Comment:
Sorry, one more thing about these UNIQUE & PRIMARY KEY column tests. Is it possible to check if these columns are still UNIQUE & PRIMARY KEY after table creation? 

I think using the ODBC functions [SQLSpecialColumns](https://www.ibm.com/docs/en/db2-for-zos/11?topic=functions-sqlspecialcolumns-get-special-row-identifier-columns) & [SQLPrimaryKeys](https://www.ibm.com/docs/en/db2-for-zos/11?topic=functions-sqlprimarykeys-get-primary-key-columns-table) will be useful in this case.

There are examples of their use in the `odbc/test_metadata.cpp` file.